state-sync/state-sync-v2/state-sync-driver/src/bootstrapper.rs

```diff
@@ -855,26 +855,6 @@ impl<
             )));
         }
 
-        // Verify the payload end index is valid
-        let expected_end_index = state_value_chunk_with_proof
```

Comment:
Where is this verification being done?

Reply:
This is being done directly above, so we can remove the additional check, i.e., both checks are logically equivalent:
The first checks if:

```
last_index - first_index + 1 == num_entries
```

and second check is:

```
first index + num_entries - 1 == last_index
```